workspaces/adoption-insights/plugins/adoption-insights/src/components/Templates/Templates.tsx

```diff
@@ -0,0 +1,164 @@
+import TableBody from '@mui/material/TableBody';
+import TableCell from '@mui/material/TableCell';
+import TableFooter from '@mui/material/TableFooter';
+import TableHead from '@mui/material/TableHead';
+import TableRow from '@mui/material/TableRow';
+import CircularProgress from '@mui/material/CircularProgress';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import { entityRouteRef } from '@backstage/plugin-catalog-react';
+import { useRouteRef } from '@backstage/core-plugin-api';
+import Link from '@mui/material/Link';
+
+import CardWrapper from '../CardWrapper';
+import { TEMPLATE_TABLE_HEADERS } from '../../utils/constants';
+import TableFooterPagination from '../CardFooter';
+import { useTemplates } from '../../hooks/useTemplates';
+
+const Templates = () => {
+  const [page, setPage] = React.useState(0);
+  const [limit] = React.useState(20);
+  const [rowsPerPage, setRowsPerPage] = React.useState(3);
+  const entityLink = useRouteRef(entityRouteRef);
+
+  const { templates, loading } = useTemplates({ limit });
+
+  const handleChangePage = React.useCallback(
+    (_event: unknown, newPage: number) => {
+      setPage(newPage);
+    },
+    [],
+  );
+
+  const handleChangeRowsPerPage = React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+      setRowsPerPage(+event.target.value);
+      setPage(0);
+    },
+    [],
+  );
+
+  const visibleTemplates = React.useMemo(() => {
+    return templates.data?.slice(
+      page * rowsPerPage,
+      page * rowsPerPage + rowsPerPage,
+    );
+  }, [templates, page, rowsPerPage]);
+
+  if (!visibleTemplates || visibleTemplates?.length === 0) {
+    return (
+      <CardWrapper title="Top templates">
+        <Box
+          display="flex"
+          justifyContent="center"
+          alignItems="center"
+          height={200}
+        >
+          <Typography>No results for this time range.</Typography>
+        </Box>
+      </CardWrapper>
+    );
+  }
+
+  return (
+    <CardWrapper title={`Top ${rowsPerPage} templates`}>
+      <Table aria-labelledby="Catalog entities" sx={{ width: '100%' }}>
+        <TableHead>
+          <TableRow>
+            {TEMPLATE_TABLE_HEADERS.map(header => (
+              <TableCell
+                key={header.id}
+                align="left"
+                sx={{
+                  borderBottom: theme => `1px solid ${theme.palette.grey[300]}`,
+                }}
+              >
+                {header.title}
+              </TableCell>
+            ))}
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {loading ? (
+            <TableRow>
+              <TableCell colSpan={TEMPLATE_TABLE_HEADERS.length} align="center">
+                <CircularProgress />
+              </TableCell>
+            </TableRow>
+          ) : (
+            visibleTemplates?.map(template => (
+              <TableRow
+                key={template.entityRef}
```

Comment:
I have changed this to lowercase in the response, Please rebase with main and align with that changes.
```suggestion
                key={template.entityref}
```